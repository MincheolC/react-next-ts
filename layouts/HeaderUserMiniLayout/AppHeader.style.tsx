import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import grey from '@material-ui/core/colors/grey'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    appBar: {
      boxShadow: 'none',
      maxHeight: '64px',
      borderBottom: `1px solid ${grey[300]}`,
      padding: '0 55px',
    },
    toolBar: {
      display: 'flex',
      justifyContent: 'space-between',
    },
    appTitle: {
      fontSize: Math.round(theme.typography.fontSize * 1.5),
      fontWeight: theme.typography.fontWeightBold,
    },
    menuContainer: {
      display: 'flex',
      alignItems: 'center',
      '& > a': {
        margin: '0 6px',
        '&:first-of-type': {
          marginLeft: '20px',
        },
      },
    },
    menu: {
      fontSize: Math.round(theme.typography.fontSize),
      fontWeight: theme.typography.fontWeightBold,
      margin: '0 10px',
    },
  })
)

export default useStyles
