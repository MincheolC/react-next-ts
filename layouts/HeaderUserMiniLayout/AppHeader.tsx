/* eslint-disable react/jsx-no-target-blank */
import Image from 'next/image'
import React from 'react'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Box from '@material-ui/core/Box'
import GitHubIcon from '@material-ui/icons/GitHub'
import useStyles from './AppHeader.style'

interface AppHeaderProps {
  title: string
  menu: string[]
}

const AppHeader: React.FC<AppHeaderProps> = (props) => {
  const classes = useStyles({ lastMenuIndex: props.menu.length })

  return (
    <>
      <AppBar className={classes.appBar} color="inherit">
        <Toolbar className={classes.toolBar}>
          <Box className={classes.appTitle}>{props.title}</Box>
          <Box className={classes.menuContainer}>
            {props.menu.map((name) => (
              <Box key={name} className={classes.menu} component="span">
                {name}
              </Box>
            ))}
            <a href="https://github.com/MincheolC" target="_blank">
              <GitHubIcon style={{ fontSize: 30 }} />
            </a>
            <a
              href="https://www.notion.so/Charles-Dev-b2c65d1ae5ee4df99d4e24abe37eb38a"
              target="_blank"
            >
              <Image
                src="/images/notion-icon-96-black.png"
                width={31}
                height={31}
              />
            </a>
          </Box>
        </Toolbar>
      </AppBar>
    </>
  )
}

export default AppHeader
