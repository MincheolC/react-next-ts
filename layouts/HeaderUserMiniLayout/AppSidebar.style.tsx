import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    sidebar: {
      zIndex: 100,
      width: '15rem',
      transition: 'all 0.5s ease',

      [theme.breakpoints.up('lg')]: {
        width: '4rem',
        position: 'fixed',
        left: 0,
      },

      [theme.breakpoints.up('xl')]: {
        width: '5rem',
      },
    },
    sidebarBg: {
      backgroundColor: '#313541',
    },
    sidebarStandard: {
      height: '100%',
      width: '100%',
      color: 'white',
      overflow: 'hidden',
    },
  })
)

export default useStyles
