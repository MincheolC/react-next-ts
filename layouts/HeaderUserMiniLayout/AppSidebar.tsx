import React, { useContext } from 'react'
import clsx from 'clsx'
import Drawer from '@material-ui/core/Drawer'
import Box from '@material-ui/core/Box'
import useStyles from './AppSidebar.style'
import { AppContext } from '../../components/utility/ContextProvider'

interface AppSidebarProps {
  variant?: string
  isCollapsed?: boolean
  setCollapsed?: () => boolean
}

const AppSidebar: React.FC<AppSidebarProps> = ({ variant = '' }) => {
  const classes = useStyles()
  const { navCollapsed, changeNavCollapsed } = useContext(AppContext)

  return (
    <>
      <Drawer
        classes={{
          root: clsx(variant),
          paper: clsx(variant),
        }}
        open={navCollapsed}
        onClose={() => changeNavCollapsed()}
      >
        <Box height="100%" className={clsx(classes.sidebar, 'app-sidebar')}>
          <Box className={clsx(classes.sidebarBg, classes.sidebarStandard)}>
            Hello
          </Box>
        </Box>
      </Drawer>
    </>
  )
}

export default AppSidebar
