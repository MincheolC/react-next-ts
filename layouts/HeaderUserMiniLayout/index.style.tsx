import { createStyles, makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(() =>
  createStyles({
    appMain: {
      height: '100vh',
      position: 'relative',
    },
    mainContent: {
      flex: 1,
      paddingTop: '140px',
      display: 'flex',
      justifyContent: 'space-between',
    },
    mainContainer: {
      width: '100%',
    },
  })
)

export default useStyles
