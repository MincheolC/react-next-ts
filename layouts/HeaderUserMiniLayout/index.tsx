import React from 'react'
import Box from '@material-ui/core/Box'
import AppHeader from './AppHeader'
import useStyles from './index.style'

const APP_TITLE = 'CHARLES DEV'
const MENU = [
  { name: 'REACT', url: '/' },
  { name: 'CSS', url: '/css' },
  { name: 'FLUTTER', url: '/' },
  { name: 'TYPESCRIPT', url: '/' },
]

const HeaderUserMiniLayout: React.FC<React.ReactNode> = (props) => {
  const classes = useStyles()

  return (
    <Box className={classes.appMain}>
      <AppHeader title={APP_TITLE} menu={MENU} />
      <Box className={classes.mainContent}>
        <Box className={classes.mainContainer}>{props.children}</Box>
      </Box>
    </Box>
  )
}

export default HeaderUserMiniLayout
