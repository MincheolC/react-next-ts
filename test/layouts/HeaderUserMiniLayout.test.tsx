import React from 'react'
import { render, screen } from '../testUtils'
import HeaderUserMiniLayout from '../../layouts/HeaderUserMiniLayout'

describe('HeaderUserMiniLayout', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HeaderUserMiniLayout />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('navigators', () => {
    render(<HeaderUserMiniLayout />)
    expect(screen.getByText(/css/i).getAttribute('href')).toBe('/css')
    expect(screen.getByText(/react/i).getAttribute('href')).toBe('/')
    expect(screen.getByText(/flutter/i).getAttribute('href')).toBe('/')
    expect(screen.getByText(/typescript/i).getAttribute('href')).toBe('/')

    expect(screen.getByTestId('myNotion').getAttribute('href')).toBe(
      'https://www.notion.so/Charles-Dev-b2c65d1ae5ee4df99d4e24abe37eb38a'
    )
    expect(screen.getByTestId('myGithub').getAttribute('href')).toBe(
      'https://github.com/MincheolC'
    )
  })
})
