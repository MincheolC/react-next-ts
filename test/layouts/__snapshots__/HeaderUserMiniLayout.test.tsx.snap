// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUserMiniLayout matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-appMain-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorInherit makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-6 MuiToolbar-gutters"
      >
        <div
          class="MuiBox-root MuiBox-root-10 makeStyles-appTitle-7"
        >
          CHARLES DEV
        </div>
        <div
          class="MuiBox-root MuiBox-root-11 makeStyles-menuContainer-8"
        >
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            REACT
          </a>
          <a
            class="makeStyles-menu-9"
            href="/css"
          >
            CSS
          </a>
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            FLUTTER
          </a>
          <a
            class="makeStyles-menu-9"
            href="/"
          >
            TYPESCRIPT
          </a>
          <a
            data-testid="myGithub"
            href="https://github.com/MincheolC"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 30px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </a>
          <a
            data-testid="myNotion"
            href="https://www.notion.so/Charles-Dev-b2c65d1ae5ee4df99d4e24abe37eb38a"
            target="_blank"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzEiIGhlaWdodD0iMzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </a>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root MuiBox-root-12 makeStyles-mainContent-2"
    >
      <div
        class="MuiBox-root MuiBox-root-13 makeStyles-mainContainer-3"
      />
    </div>
  </div>
</DocumentFragment>
`;
